src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/PackageReferenceCommands/ListPackage/ListPackageCommandRunner.cs

```diff
@@ -224,6 +256,63 @@ public async Task ExecuteCommandAsync(ListPackageArgs listPackageArgs)
             GetVersionsFromDict(packages, packagesVersionsDict, listPackageArgs);
         }
 
+        /// <summary>
+        /// Fetches deprecation info for all of the packages found in a project.
+        /// </summary>
+        /// <param name="packages">The packages found in a project.</param>
+        /// <param name="listPackageArgs">List args for the token and source provider</param>
+        private async Task GetDeprecationInfoAsync(
+            IEnumerable<FrameworkPackages> packages,
+            ListPackageArgs listPackageArgs)
+        {
+            var providers = Repository.Provider.GetCoreV3();
```

Comment:
Feels like quite a bit of duplication with the outdated logic.

Reply:
See https://github.com/NuGet/NuGet.Client/pull/2959/commits/122c990fb3fa8b8aeb13f79a848798498eed582a